File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,ArrayList<String>> anagrams = new HashMap<>();
        List<List<String>> result = new ArrayList<>();
        for(String str: strs)
        {
            char[] ch = str.toCharArray();
            Arrays.sort(ch);
            anagrams.computeIfAbsent(new String(ch),k -> new ArrayList<>()).add(str);
        }

        for(ArrayList<String> val : anagrams.values() )
        {
            result.add(val);
        }

        return result;

    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        map = {}

        for str in strs:
            char_sorted = sorted(str)
            sorted_str = "".join(char_sorted)

            if sorted_str in map:
                map[sorted_str].append(str)
            
            else:
                map[sorted_str] = [str]
        

        result =  map.values()
        return list(result)